systempoweraction/src/main/java/com/stream_pi/systempoweraction/SystemPowerAction.java

```diff
@@ -0,0 +1,196 @@
+        setAuthor("quimodotcom");
+        setHelpLink("https://github.com/quimodotcom/essential-actions");
+        setVersion(new Version(1,0,0));
+        setCategory("System");
+        
+        states = new ArrayList<>();
+        states.addAll(Arrays.asList(
+                new ListValue("Shutdown"),
+                new ListValue("Restart"),
+                new ListValue("Sleep")
+        ));
+    }
+    
+    private final ArrayList<ListValue> states;
+    
+    @Override
+    public void initProperties() throws MinorException
+    {
+        Property status = new Property("state", Type.LIST);
+        status.setListValue(states);
+        status.setDisplayName("Power State");
+        
+        Property customLinuxSleep = new Property("linuxsl", Type.STRING);
+        customLinuxSleep.setDisplayName("Custom Linux Sleep Command");
+        customLinuxSleep.setCanBeBlank(true);
+        
+        Property customLinuxRestart = new Property("linuxre", Type.STRING);
+        customLinuxRestart.setDisplayName("Custom Linux Restart Command");
+        customLinuxRestart.setCanBeBlank(true);
+
+        Property customLinuxShutdown = new Property("linuxsh", Type.STRING);
+        customLinuxShutdown.setDisplayName("Custom Linux Shutdown Command");
+        customLinuxShutdown.setCanBeBlank(true);
+        
+        addClientProperties(status);
+        addServerProperties(customLinuxSleep, customLinuxRestart, customLinuxShutdown);
+    }
+    
+    @Override
+    public void onActionClicked() throws MinorException
+    {
+        String state = states.get(getClientProperties().getSingleProperty("state").getSelectedIndex()).getName().toString();
+        switch (state)
+        {
+            case "Shutdown":
+                shutdown();
+                break;
+            
+            case "Restart":
+                restart();
+                break;
+                
+            case "Sleep":
+                sleep();
+                break;     
+        }
+    }
+    
+    public void shutdown() throws MinorException
+    {
+        String shutdownCommand = "";
+        try{
+            switch (getPlatform()) {
+            case MAC:
+                    Process p = Runtime.getRuntime().exec
+                         ("/bin/bash");
+                     String command = "osascript -e 'tell application \"System Events\"' "
+                             + " -e \"shut down\" -e 'end tell'";
+                     OutputStream stdin = p.getOutputStream();
+                     stdin.write( command.getBytes() );
+                     stdin.flush();
+                     stdin.close();
+                break;
+            case WINDOWS:
+                shutdownCommand = "shutdown.exe -s -t 0";
+                Runtime.getRuntime().exec(shutdownCommand);
+                break;
+            case LINUX:
+                String shutdown = getClientProperties().getSingleProperty("linuxsh").getStringValue();
+                if(shutdown != "")
```

Comment:
Instead of this, go with the setDefaultValueStr, more details written in the first comment.

Reply:
no point